crates/astria-conductor/src/executor/mod.rs

```diff
@@ -695,32 +695,41 @@ impl ExecutableBlock {
     }
 }
 
+#[instrument(skip_all)]
 fn insert_price_feed_data_into_transactions_if_exists(
     transactions: Vec<Bytes>,
     extended_commit_info: Option<ExtendedCommitInfoWithCurrencyPairMapping>,
 ) -> Vec<Bytes> {
+    use astria_core::oracles::price_feed::utils::calculate_prices_from_vote_extensions;
     use prost::Message as _;
 
-    if let Some(extended_commit_info) = extended_commit_info {
-        let Ok(prices) =
-            astria_core::oracles::price_feed::utils::calculate_prices_from_vote_extensions(
-                &extended_commit_info.extended_commit_info,
-                &extended_commit_info.id_to_currency_pair,
-            )
-        else {
-            debug!(
+    let Some(extended_commit_info) = extended_commit_info else {
+        return transactions;
+    };
+
+    let prices = match calculate_prices_from_vote_extensions(
+        &extended_commit_info.extended_commit_info,
+        &extended_commit_info.id_to_currency_pair,
+    ) {
+        Ok(prices) => prices,
+        Err(error) => {
+            warn!(
+                error = %error,
                 "failed to calculate prices from vote extensions; continuing without price feed \
-                 data"
+                data"
             );
             return transactions;
-        };
-        let rollup_data = RollupData::PriceFeedData(Box::new(PriceFeedData::new(prices)));
-        std::iter::once(rollup_data.into_raw().encode_to_vec().into())
-            .chain(transactions)
-            .collect()
-    } else {
-        transactions
-    }
+        }
+    };
+
+    // Prepend the price data to the transactions. Regardless of the order here, the rollup can
+    // choose whichever execution order suits best for its use case, but it's anticipated that
+    // applying the updated price feed before executing transactions would be a common use case, so
+    // the price feed is prepended for convenience.
+    let rollup_data = RollupData::PriceFeedData(Box::new(PriceFeedData::new(prices)));
+    std::iter::once(rollup_data.into_raw().encode_to_vec().into())
```

Comment:
Irrespective of whether we chain or do push + rotate, can you move this to a freestanding function, i.e. smth like `fn prepend<T>(xs: Vec<T>, x: T)`?